src/zenml/cli/base.py

```diff
@@ -64,18 +67,31 @@ def init(path: Optional[Path]) -> None:
 
 
 @cli.command("clean")
-@click.option("--yes", "-y", type=click.BOOL, default=False)
+@click.option("--yes", "-y", is_flag=True, default=False)
 def clean(yes: bool = False) -> None:
-    """Clean everything in repository.
+    """Delete all ZenML metadata and artifacts.
+
+    This is a destructive operation, primarily intended for use in development.
 
     Args:
       yes: bool:  (Default value = False)
     """
     if not yes:
-        _ = confirmation(
-            "This will completely delete all pipelines, their associated "
+        confirm = confirmation(
+            "DANGER: This will completely delete all pipelines, their associated "
             "artifacts and metadata ever created in this ZenML repository. "
             "Are you sure you want to proceed?"
         )
 
-    error("Not implemented for this version")
+    if yes or confirm:
+        local_zen_repo_config = Path.cwd() / REPOSITORY_DIRECTORY_NAME
+        global_zen_config = get_global_config_directory()
+        if exists(str(local_zen_repo_config)):
```

Comment:
It feels a little strange to me to have `exists`/`rmtree` without the `fileio.` prefix similar to (most of) the remaining code, but feel free to ignore